src/components/shared/ComponentEditor/ComponentEditorDialog.tsx

```diff
@@ -0,0 +1,197 @@
+  return (
+    <FullscreenElement fullscreen={true}>
+      <BlockStack className="h-full w-full p-2 bg-white" align="start" gap="2">
+        <InlineStack
+          className="w-full h-13"
+          gap="4"
+          blockAlign="center"
+          align="space-between"
+          wrap="nowrap"
+        >
+          <Skeleton size="lg" />
+          <Skeleton size="lg" shape="button" />
+        </InlineStack>
+        <Separator />
+        <InlineStack
+          className="w-full h-full flex-1"
+          gap="4"
+          blockAlign="start"
+          align="space-between"
+          wrap="nowrap"
+        >
+          <BlockStack gap="2" align="start" className="flex-1">
+            <Skeleton size="full" />
+            <Skeleton size="half" />
+            <Skeleton size="full" />
+          </BlockStack>
+          <BlockStack gap="2" align="start" className="flex-1">
+            <Skeleton size="full" />
+            <Skeleton size="half" />
+            <Skeleton size="full" />
+          </BlockStack>
+        </InlineStack>
+      </BlockStack>
+    </FullscreenElement>
+  );
+};
+
+type PythonCodeDetection =
+  | { isPython: true; pythonOriginalCode: string }
+  | { isPython: false };
+
+export const ComponentEditorDialog = withSuspenseWrapper(
+  ({
+    text,
+    templateName = "empty",
+    onClose,
+  }: {
+    text?: string;
+    templateName?: string;
+    onClose: () => void;
+  }) => {
+    const notify = useToastNotification();
+    const { addToComponentLibrary } = useComponentLibrary();
+
+    const { data: templateCode } = useTemplateCodeByName(templateName);
+    const [componentText, setComponentText] = useState(text ?? templateCode);
+
+    const { data: pythonCodeDetection } = useSuspenseQuery({
+      queryKey: ["isPython", `${templateName}-${JSON.stringify(text)}`],
+      queryFn: async (): Promise<PythonCodeDetection> => {
+        if (text) {
+          const hydratedComponent = await hydrateComponentReference({
+            text,
+          });
+
+          if (
+            !hydratedComponent?.spec?.metadata?.annotations
+              ?.python_original_code
+          ) {
+            return { isPython: false };
+          }
+
+          return {
+            isPython: true,
+            pythonOriginalCode: hydratedComponent?.spec?.metadata?.annotations
+              ?.python_original_code as string,
+          };
+        }
+
+        if (templateName !== "python") {
+          return { isPython: false };
+        }
+
+        const defaultPythonFunctionText = await import(
+          `./templates/python_function.py?raw`
+        ).then((module) => module.default);
+
+        return {
+          isPython: true,
+          pythonOriginalCode: defaultPythonFunctionText,
+        };
+      },
+    });
+
+    const handleComponentTextChange = useCallback(
+      (value: string | undefined) => {
+        setComponentText(value ?? "");
+      },
+      [],
+    );
```

Comment:
If I've correctly read my way through the call stack, I don't think this should be `undefined`


```suggestion
    const handleComponentTextChange = useCallback(
      (value: string) => {
        setComponentText(value);
      },
      [],
    );
```